src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -503,7 +503,7 @@ boolean canRepair(
 
     if (RepairSegment.State.NOT_STARTED == segment.getState()) {
       try {
-        Map<String, String> dcByNode = getDCsByNodeForRepairSegment(coordinator, cluster, segment, keyspace);
+        Map<String, String> dcByNode = segment.getReplicas();
```

Comment:
nice that this is reducing the complexity (time and network calls) involved when determining whether to start a segment!